blivet/threads.py

```diff
@@ -33,7 +34,7 @@
 
 def exclusive(m):
     """ Run a callable while holding the global lock. """
-    @wraps(m)
+    @wraps(m, set(functools.WRAPPER_ASSIGNMENTS) & set(dir(m)))
```

Comment:
Before I added the additional arguments I saw errors like this when running under python2:

```
[dlehman@tsuga blivet]$ PYTHONPATH=.:tests:../pykickstart/ python2 -m unittest -v -f devices_test.dependencies_test
Traceback (most recent call last):
  File "/usr/lib64/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib64/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/usr/lib64/python2.7/unittest/__main__.py", line 12, in <module>
    main(module=None)
  File "/usr/lib64/python2.7/unittest/main.py", line 94, in __init__
    self.parseArgs(argv)
  File "/usr/lib64/python2.7/unittest/main.py", line 149, in parseArgs
    self.createTests()
  File "/usr/lib64/python2.7/unittest/main.py", line 158, in createTests
    self.module)
  File "/usr/lib64/python2.7/unittest/loader.py", line 130, in loadTestsFromNames
    suites = [self.loadTestsFromName(name, module) for name in names]
  File "/usr/lib64/python2.7/unittest/loader.py", line 91, in loadTestsFromName
    module = __import__('.'.join(parts_copy))
  File "/home/dlehman/src/blivet/tests/devices_test/dependencies_test.py", line 5, in <module>
    from blivet.deviceaction import ActionCreateDevice
  File "blivet/deviceaction.py", line 28, in <module>
    from .devices import StorageDevice
  File "blivet/devices/__init__.py", line 23, in <module>
    from .device import Device
  File "blivet/devices/device.py", line 37, in <module>
    class Device(util.ObjectID):
  File "/usr/lib/python2.7/site-packages/six.py", line 824, in wrapper
    return metaclass(cls.__name__, cls.__bases__, orig_vars)
  File "blivet/threads.py", line 77, in __new__
    obj = property(fget=exclusive(obj.__get__),
  File "blivet/threads.py", line 45, in exclusive
    @wraps(m)
  File "/usr/lib/python2.7/site-packages/six.py", line 792, in wrapper
    f = functools.wraps(wrapped, assigned, updated)(f)
  File "/usr/lib64/python2.7/functools.py", line 33, in update_wrapper
    setattr(wrapper, attr, getattr(wrapped, attr))
AttributeError: 'method-wrapper' object has no attribute '__module__'
[dlehman@tsuga blivet]$ 
```